protostar/utils/protostar_directory.py

```diff
@@ -58,30 +63,33 @@ def protostar_version(self) -> Optional[VersionType]:
         )
         try:
             with open(path, "r", encoding="UTF-8") as file:
-                version_s = tomli.loads(file.read())["tool"]["poetry"]["version"]
-                return VersionManager.parse(version_s)
-        except FileNotFoundError:
-            getLogger().warning("Couldn't read Protostar version")
-            return None
+                self._pyproject_toml_dict = tomli.loads(file.read())
+                return self._pyproject_toml_dict
+        except FileNotFoundError as err:
+            raise ProtostarException("Couldn't read Protostar version") from err
 
     @property
-    def cairo_version(self) -> Optional[VersionType]:
-        path = (
-            self._protostar_directory.directory_root_path
-            / "dist"
-            / "protostar"
-            / "info"
-            / "pyproject.toml"
-        )
-        try:
-            with open(path, "r", encoding="UTF-8") as file:
-                version_s = tomli.loads(file.read())["tool"]["poetry"]["dependencies"][
-                    "cairo-lang"
-                ]
-                return VersionManager.parse(version_s)
-        except FileNotFoundError:
-            getLogger().warning("Couldn't read cairo-lang version")
-            return None
+    def protostar_version(self) -> VersionType:
+        version_s = self.pyproject_toml["tool"]["poetry"]["version"]
+        return VersionManager.parse(version_s)
+
+    @property
+    def cairo_version(self) -> VersionType:
+        version_s = self.pyproject_toml["tool"]["poetry"]["dependencies"]["cairo-lang"]
+        return VersionManager.parse(version_s)
+
+    @property
+    def toml_breaking_versions(self) -> List[VersionType]:
+        return [
+            VersionManager.parse(v)
+            for v in self.pyproject_toml["tool"]["protostar"]["toml_breaking_versions"]
+        ]
+
+    def version_range_has_breaking_changes(self, low: VersionType, high: VersionType):
```

Comment:
Methods and functions should start with a noun.

Reply:
this does start with a noun

Reply:
He probably meant verb.

Renaming it to `has_version_range_breaking_changes` won't hurt imho

Reply:
Also please add `-> bool` here